xml/s4s_tune_wmp.xml

```diff
@@ -0,0 +1,519 @@
+                    </important>
+                    <para>The following message can be ignored:</para>
+                    <screen>Warning: Found memory controller on v1<!--
+                       --> hierarchy. Make sure unified hierarchy only is<!--
+                       --> used.</screen>
+                    <para>Switching to unified hierarchy is done in the next step.</para>
+                </step>
+                <step>
+                    <para>Add <varname>systemd.unified_cgroup_hierarchy=true</varname> to the
+                        kernel command line by adding it to
+                        <varname>GRUB_CMDLINE_LINUX_DEFAULT</varname> in <filename>/etc/default/grub</filename>
+                        like:</para>
+                    <screen>GRUB_CMDLINE_LINUX_DEFAULT="... systemd.unified_cgroup_hierarchy=true swapaccount=1"</screen>
+                    <para>The parameter <varname>swapaccount=1</varname> is not needed for
+                        WMP to work, but it aids the analysis in support
+                        cases to show the amount of swapped out memory for
+                        each cgroup.</para>
+                </step>
+                <step>
+                    <para>Rewrite the GRUB2 configuration:</para>
+                    <screen>&prompt.sudo;<command>grub2-mkconfig</command> -o /boot/grub2/grub.cfg</screen>
+                    <para>After reboot (will be done later), the cgroup
+                        hierarchy is switched to v2 (unified hierarchy)
+                        only.</para>
+                </step>
+                <step>
+                    <para>Configure MemoryLow for the <systemitem>SAP.slice</systemitem>:</para>
+                    <screen>&prompt.sudo;<command>systemctl</command> set-property SAP.slice MemoryLow=...</screen>
+                    <para>
+                        This command creates a drop-in in
+                            <filename>/etc/systemd/system.control/SAP.slice.d/</filename> to set
+                            MemoryLow.
+                    </para>
+                    <note>
+                        <title>Correctly Calculate MemoryLow Value</title>
+                        <para>
+                        The value for MemoryLow is workload specific.
+                        Read the &sap; Note <link
+                         xlink:href="https://launchpad.support.sap.com/#/notes/1557506"
+                         >1557506 - Linux paging improvements</link> for
+                        how to calculate the correct value.
+                        </para>
+                        <para>MemoryLow takes the memory size in bytes. If
+                        the value is suffixed with K, M, G or T, the
+                        specified memory size is parsed as Kilobytes,
+                        Megabytes, Gigabytes, or Terabytes (with the base
+                        1024), respectively. Alternatively, a percentage
```

Comment:
Good point! What do you think @scmschmidt?

Reply:
In principle I would agree, but this comes directly from the man page of systemd.resource-control(5):

      MemoryLow=bytes
           ...
           Takes a memory size in bytes. If the value is suffixed with K, M, G or T, the specified memory size is parsed as
           Kilobytes, Megabytes, Gigabytes, or Terabytes (with the base 1024), respectively. Alternatively, a percentage value
           may be specified, which is taken....

Reply:
Hmn, :thinking: that's a tricky question. Maybe we should leave it as it is. Deviating from the manpage could confuse people. At least the base is written (1024 instead of 1000) which should be enough, although the units are not entirely correct.

@lproven what do you think?

Reply:
On second thought, I think we should correct it. We don't have to do the same "mistakes" as the systemd guys. ;-)
I just copied it to make my life easier.

Reply:
Okay, you've convinced me. :wink:  I've changed that to:

```
MemoryLow takes the memory size in bytes. If
the value is suffixed with K, M, G or T, the
specified memory size is parsed as Kibibytes,
Mebibytes, Gibibytes, or Terabytes (with the base
1024 instead of 1000, see <link
  xlink:href="https://en.wikipedia.org/wiki/Binary_prefix"/>),
respectively.
```